src/oskbd/linux.rs

```diff
@@ -435,40 +444,67 @@ impl KbdOut {
         self.release_key(btn.into())
     }
 
-    pub fn scroll(&mut self, direction: MWheelDirection, distance: u16) -> Result<(), io::Error> {
-        log::debug!("scroll: {direction:?} {distance:?}");
+    pub fn scroll(
+        &mut self,
+        direction: MWheelDirection,
+        hi_res_distance: u16,
+    ) -> Result<(), io::Error> {
+        log::debug!("scroll: {direction:?} {hi_res_distance:?}");
+
+        let mut lo_res_distance = hi_res_distance / HI_RES_SCROLL_UNITS_IN_LO_RES;
+        let leftover_hi_res_distance = hi_res_distance % HI_RES_SCROLL_UNITS_IN_LO_RES;
+
         match direction {
             MWheelDirection::Up | MWheelDirection::Down => {
-                let lo_res_distance = distance / HI_RES_SCROLL_UNITS_IN_LO_RES;
-                if lo_res_distance > 0 {
-                    self.do_scroll(direction, lo_res_distance)?;
-                }
-                let leftover_scroll = distance % HI_RES_SCROLL_UNITS_IN_LO_RES;
-                if leftover_scroll > 0 {
-                    self.accumulated_scroll += leftover_scroll;
-                    if self.accumulated_scroll >= HI_RES_SCROLL_UNITS_IN_LO_RES {
-                        self.accumulated_scroll -= HI_RES_SCROLL_UNITS_IN_LO_RES;
-                        self.do_scroll(direction, 1)?;
-                    }
-                }
+                self.accumulated_scroll += leftover_hi_res_distance;
+                lo_res_distance += self.accumulated_scroll % HI_RES_SCROLL_UNITS_IN_LO_RES;
```

Comment:
This seems to be the wrong operation - it should be `/`. In my testing, the `write` branch (hi-res only) never executes, and the branch that executes is always `write_many` because of this.

Reply:
You're right! I just updated my branch to fix it.